clients/githubrepo/tarball.go

```diff
@@ -80,6 +83,7 @@ func (handler *tarballHandler) init(ctx context.Context, repo *github.Repository
 	handler.once = new(sync.Once)
 	handler.ctx = ctx
 	handler.repo = repo
+	handler.httpClient = makeHTTPClient(ctx)
```

Comment:
Can simply pass in `httpClient` here - https://github.com/ossf/scorecard/blob/main/clients/githubrepo/client.go#L233: 

```
httpClient := ...
return &Client {
  tarball: tarballHandler{
    httpClient: httpClient,
  },
  ...
}
```